pkg/interval/intervalset.go

```diff
@@ -2,36 +2,52 @@ package interval
 
 import (
 	"fmt"
+	"log"
 	"slices"
 	"sort"
 )
 
 // CanonicalSet is a canonical representation of a set of Interval objects
 type CanonicalSet struct {
-	IntervalSet []Interval // sorted list of non-overlapping intervals
+	intervalSet []Interval // sorted list of non-overlapping intervals
 }
 
 func NewCanonicalIntervalSet() *CanonicalSet {
 	return &CanonicalSet{
-		IntervalSet: []Interval{},
+		intervalSet: []Interval{},
 	}
 }
 
+func (c *CanonicalSet) Intervals() []Interval {
+	return slices.Clone(c.intervalSet)
+}
+
+func (c *CanonicalSet) NumIntervals() int {
+	return len(c.intervalSet)
+}
+
+func (c *CanonicalSet) Min() int64 {
+	if len(c.intervalSet) == 0 {
+		log.Fatal("cannot take min from empty interval set")
```

Comment:
Better use `log.Panic()`, allowing the caller to recover or to set the exit code as it likes

Reply:
Done